File: articles/grit-and-academic-persistence/fortran/grit_academic_persistence_score.f90
program grit_academic_persistence_score
  implicit none
  real :: perseverance_effort, consistency_interests, grit
  real :: self_control, belonging, social_support, financial_stress, burnout
  real :: readiness

  perseverance_effort = 0.82
  consistency_interests = 0.64
  self_control = 0.72
  belonging = 0.74
  social_support = 0.58
  financial_stress = -0.42
  burnout = -0.30

  grit = 0.60 * perseverance_effort + 0.40 * consistency_interests
  readiness = 0.22 * grit + 0.18 * self_control + 0.22 * belonging + &
              0.18 * social_support - 0.20 * financial_stress - 0.18 * burnout

  print *, "Synthetic grit score:", grit
  print *, "Synthetic academic persistence readiness index:", readiness
end program grit_academic_persistence_score

File: articles/grit-and-academic-persistence/go/grit_academic_persistence_score.go
package main

import "fmt"

func GritScore(perseveranceEffort, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func PersistenceReadinessIndex(
	grit,
	selfControl,
	belonging,
	socialSupport,
	financialStress,
	burnout float64,
) float64 {
	return 0.22*grit +
		0.18*selfControl +
		0.22*belonging +
		0.18*socialSupport -
		0.20*financialStress -
		0.18*burnout
}

func main() {
	grit := GritScore(0.82, 0.64)
	readiness := PersistenceReadinessIndex(grit, 0.72, 0.74, 0.58, -0.42, -0.30)

	fmt.Printf("Synthetic grit score: %.3f\n", grit)
	fmt.Printf("Synthetic academic persistence readiness index: %.3f\n", readiness)
}
